dbms/src/Flash/tests/gtest_join_executor.cpp

```diff
@@ -401,20 +401,110 @@ try
 }
 CATCH
 
+TEST_F(JoinExecutorTestRunner, CrossJoinWithCondition)
+try
+{
+    context.addMockTable("cross_join", "t1", {{"a", TiDB::TP::TypeString}, {"b", TiDB::TP::TypeString}}, {toNullableVec<String>("a", {"1", "2", {}, "1"}), toNullableVec<String>("b", {"3", "4", "3", {}})});
+    context.addMockTable("cross_join", "t2", {{"a", TiDB::TP::TypeString}, {"b", TiDB::TP::TypeString}}, {toNullableVec<String>("a", {"1", "3", {}, "2"}), toNullableVec<String>("b", {"3", "4", "3", {}})});
+
+    const auto cond = gt(col("a"), lit(Field("1", 1)));
```

Comment:
I wonder which table the a belongs to, or both?
```
mysql> CREATE TABLE `test`.`t1` (
    ->   `a` varchar(20) DEFAULT NULL,
    ->   `b` varchar(20) DEFAULT NULL
    -> );
Query OK, 0 rows affected (0.10 sec)

mysql> CREATE TABLE `test`.`t2` (
    ->   `a` varchar(20) DEFAULT NULL,
    ->   `b` varchar(20) DEFAULT NULL
    -> );
Query OK, 0 rows affected (0.08 sec)

mysql> insert into test.t1 values ('1', '3'), ('2', '4'), (NULL, '3'), ('1', NULL);
Query OK, 4 rows affected (0.00 sec)
Records: 4  Duplicates: 0  Warnings: 0

mysql> 
mysql> insert into test.t2 values ('1', '3'), ('2', '4'), (NULL, '3'), ('1', NULL);
Query OK, 4 rows affected (0.00 sec)
Records: 4  Duplicates: 0  Warnings: 0

mysql> 
mysql> select * from test.t1 join test.t2 on a > '1';
ERROR 1052 (23000): Column 'a' in field list is ambiguous
```

Reply:
Since our framework doesn't support in using qualified name for table column yet, current implementation is to choose the first column with the same name.

Reply:
PTAL current comments